build-scripts/components/kubernetes/version.sh

```diff
@@ -3,6 +3,10 @@
 KUBE_TRACK="${KUBE_TRACK:-1.22}"            # example: "1.24"
 KUBE_VERSION="${KUBE_VERSION:-}"        # example: "v1.24.2"
 
+if [ -e "${SNAPCRAFT_STAGE}/KUBE_VERSION" ]; then
+  KUBE_VERSION=$(cat "$SNAPCRAFT_STAGE/KUBE_VERSION")
+fi
+
 if [ -z "${KUBE_VERSION}" ]; then
   if [ -z "${KUBE_TRACK}" ]; then
     KUBE_VERSION="$(curl -L --silent "https://dl.k8s.io/release/stable.txt")"
```

Comment:
This is all we really need to do to set the version, no need to mess with passing version numbers around.
```suggestion
KUBE_TRACK="${KUBE_TRACK:-1-22}"            # example: "1-22"

KUBE_VERSION=$(curl -L --silent https://raw.githubusercontent.com/aws/eks-distro/main/projects/kubernetes/kubernetes/${KUBE_TRACK}/GIT_TAG)
```

Reply:
@berkayoz ^ see my suggestion, we can then cleanup the rest of the parts. similarly, you can update the `kubernetes-version` part to append the eks release suffix as needed.